processor/src/test/java/org/mapstruct/ap/test/bugs/_1482/TargetSourceMapper.java

```diff
@@ -20,9 +20,9 @@
     @Mapping(target = "wrapper", source = "bigDecimal")
     abstract Source2 map(Target target);
 
-    protected <T extends Enum<T>> Enum<T> map(String in, @TargetType  Class<T>clz ) {
+    protected <T extends Enum<T>> T map(String in, @TargetType Class<T>clz ) {
```

Comment:
Argh.. I'm just reading your old comments I missed originally..

When I changed this I think that the new solution is right. So we might need to document this (I think we posted this pattern in a number of places).

Let me try to explain:

We want to map `String` to a `SourceEnum`, right? Note: `Enum<SourceEnum>` should be an equivalent notation, and it is. Try changing the `Source2#test` to a `SourceEnum` instead of the current `Enum<SourceEnum>` and it (fortunately) works with the current setup.

That might lead you to think that you can incur on this solution. So, I tried: `Enum<Enum<SourceEnum>>` But the compiler will not allow you to do that. Error message:  `Type parameter 'java.lang.Enum' is not within its bound; should extend 'java.lang.Enum<java.lang.Enum<org.mapstruct.ap.test.bugs._1482.SourceEnum>>`. I don't understand that aspect. Perhaps some java expert can try to explain me what is wrong here. It bothered me so I posted [this ](https://stackoverflow.com/questions/66840251/why-cant-i-recur-in-a-java-java-lang-enum-generic-type-definition)question on SO.

Now, going back to the method: If we follow the pattern here, we should match a method:

So, let `T = Enum<SourceEnum>` the signature would say: `Enum<Enum<SourceEnum>> map( String in, @TargetType Class<Enum.class> clz)` which is deemed wrong by the compiler.

Instead we need `T` and hence the method looking like: `Enum<SourceEnum> map( String in, @TargetType Class<SourceEnum> clz)` and like mentioned above, it works for both `target = Enum<SourceEnum>` and `target = SourceEnum`

Reply:
SO question has been resolved.

Reply:
I understand the SO question, but I still don't understand why we need the change? The test here was due to a specific bug which you fixed with PR #1511. When working with `Enum` we are inferring the `Enum` type specifically. 

In any case, this most likely is something that no one will write.